### PR TITLE
Feature/fga/hide rust sdk

### PR DESCRIPTION
**Pull request by ganfra**
This PR handles https://github.com/vector-im/element-x-android/issues/144
There will be some more work to do later on splitting up the EventContent as it might be exposes outside of the timeline context.

### CODE REVIEW

**Review by jmartinesp — APPROVED** [review]
I think it looks good overall, once we fix the coverage. Thanks for the huge amount of work! I just have a couple of comments.

Also, does [this PR](https://github.com/vector-im/element-x-android/pull/215) conflict with your changes? If so, I'd wait until this one is merged to try to fix it.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt
Comment (ganfra) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:164]: I guess we can but its already secured by the `isSyncing`

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt
Comment (ganfra) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:186]: > I think it looks good overall, once we fix the coverage. Thanks for the huge amount of work! I just have a couple of comments.
> 
> Also, does [this PR](https://github.com/vector-im/element-x-android/pull/215) conflict with your changes? If so, I'd wait until this one is merged to try to fix it.

I'd wait for now, I've triggered a release of the rust-sdk-kotlin this morning but not sure what has changed

**Comment by ganfra** [comment]
> I think it looks good overall, once we fix the coverage. Thanks for the huge amount of work! I just have a couple of comments.
> 
> Also, does [this PR](https://github.com/vector-im/element-x-android/pull/215) conflict with your changes? If so, I'd wait until this one is merged to try to fix it.

I don't know if it conflicts with the changes as I've not checked exactly what is new from the rust-sdk. We should probably wait to merge this and investigate after.

**Comment by ganfra** [comment]
@bmarty 

> Rule 'Check code coverage of states' violated:
>   instructions covered percentage for class 'io.element.android.libraries.matrix.api.timeline.item.event.EventSendState$NotSendYet' is 0.000000, but expected minimum is 90

Can we disable the rule here?

File: features/login/impl/src/main/kotlin/io/element/android/features/login/impl/root/LoginRootPresenter.kt
Comment (bmarty) [inline features/login/impl/src/main/kotlin/io/element/android/features/login/impl/root/LoginRootPresenter.kt:74]: not from this PR, but we should not trim the password.

**Review by bmarty — APPROVED** [review]
LGTM (fast review!)

**Comment by bmarty** [comment]
> @bmarty
> 
> > Rule 'Check code coverage of states' violated:
> > instructions covered percentage for class 'io.element.android.libraries.matrix.api.timeline.item.event.EventSendState$NotSendYet' is 0.000000, but expected minimum is 90
> 
> Can we disable the rule here?

Sure: https://github.com/vector-im/element-x-android/pull/213/commits/b0e0c511796fad19828353dd57846ac27d11a077

**Pull request merged**
